specs/agents/metrics-health.md

```diff
@@ -0,0 +1,132 @@
+Therefore an event MUST be taken into account for these metrics if an attempt is made to submit it to the internal reporting queue, regardless of the success.
+Events dropped before this attempt (e.g. due to [sampling](tracing-sampling.md) or [transaction_max_spans](handling-huge-traces/tracing-spans-limit.md)) are NOT counted.
+
+The value of `agent.events.dropped.error` is an upper bound for the actual number of dropped events after entering the queue.
+For example, when a request to the Intake API fails without a response, all events within this request MUST be considered as failed.
+In reality, it is possible that e.g. half of the data was actually successfully received and forwarded by the APM server.
+When the APM server responds with an HTTP error code, the number of dropped events SHOULD be computed by subtracting the `accepted` field from the response body from the total number of inflight events of this request.
+
+The `agent.events.total` metric MUST have the label `event_type`. The possible values for `event_type` are the lowercase names of the events according to the [spec](https://github.com/elastic/apm-server/tree/main/docs/spec/v2) (e.g. `transaction`, `metricset`) or the value `undefined`.
+The `agent.events.dropped.*` metrics MUST NOT have any labels.
+
+Agents SHOULD attempt to assign the appropriate label value based on the counted event. If this would impose significant implementation overhead, the value `undefined` MUST be used instead.
+
+All event count metrics MUST be able to be disabled by the `disable_metrics` configuration option.
+
+### Event Queue Metrics
+
+The following metrics SHOULD be recorded by agents regarding the state of the internal event reporting queue:
+
+ - `agent.events.queue.min_size.pct`: The smallest queue usage in percent (range [0-1]) observed since this metric was reported the last time
+ - `agent.events.queue.max_size.pct`: The biggest queue usage in percent (range [0-1]) observed since this metric was reported the last time
+
+We capture both `min_size` and `max_size` to allow to distinguish between reasons for why the queue is becoming full:
+ * There is too much data captured / network errors: Both `min_size` and `max_size` are at `100%`
+ * The data is collected in bursts for which the queue is not correctly sized: `max_size` is at `100%`, but `min_size` is low.
+
+The queue usage can be computed based on how the agent defines the queue capacity.
+E.g. if the queue capacity is a fixed number of events, the usage can be computed based on the current number of events.
+If the queue capacity is in bytes, the usage can be computed based on the number of bytes currently occupied in the queue.
+
+If agents use multiple queues (e.g. separating `metricsets` from other events), they SHOULD expose the `agent.events.queue.*` metrics for every queue and add a `queue_name` label.
```

Comment:
Could it lead to issues if some agents don't define the `queue_name` label? I think it might but haven't thought it though. If so, we could require the `queue_name` label and if agents only have a single queue, they could set a name like `generic` or similar to indicate that this is a queue that holds multiple event types. This would also enable those agents to more easily add more queues in the future, without having to change the format of the metric.